File: src/main/java/de/whitefrog/neobase/model/annotation/Indexed.java
package de.whitefrog.neobase.model.annotation;

import java.lang.annotation.*;

/**
 * Indicates that a field should be handled by index.
 */
@Documented
@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface Indexed {
}

File: base/src/main/java/de/whitefrog/frogr/model/annotation/Indexed.kt
package de.whitefrog.frogr.model.annotation

/**
 * Indicates that a field should be handled by index.
 */

enum class IndexType {
  Default, LowerCase
}

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class Indexed(val type: IndexType = IndexType.Default)
